### PR TITLE
add ideal-image-slider@1.5.1 w/ npm auto-update

### LINKED ISSUE #8023: [Request] Add Ideal-Image-Slider
**Library name:** Ideal-Image-Slider
**Git repository url:** https://github.com/gilbitron/Ideal-Image-Slider
**npm:** https://www.npmjs.com/package/ideal-image-slider

### PR DESCRIPTION
**Pull request by kennynaoh**
PR for #8023 
@dannyxx001 please help me review this pull request, thank you.
Checklist for **Pull request** or **lib adding request issue** follows the conventions.

Note that if you are using a distribution purpose repository/package, please also provide the url and other related info like popularity of the source code repo/package.

# Profile of the lib
 * Git repository (required): https://github.com/gilbitron/Ideal-Image-Slider
 * Official website (optional, not the repository):
 * NPM package url (optional): https://www.npmjs.com/package/ideal-image-slider
 * GitHub / Bitbucket popularity (required):
   - Count of stars: 1,537
   - Count of watchers: 69
   - Count of forks: 162
 * NPM download stats (optional):
   - Downloads in the last day: 0
   - Downloads in the last week: 14
   - Downloads in the last month: 84

# Essential checklist
 * [ ] I'm the author of this library
   * [ ] I would like to add link to the page of this library on CDNJS on website / readme
 * [x] This lib was not found on cdnjs repo
 * [x] No already exist / duplicated issue and PR
 * [x] The lib has notable popularity
   * [x] More than 100 [Stars / Watchers / Forks] on [GitHub / Bitbucket]
   * [ ] More than 500 downloads stats per month on npm registry
 * [ ] Project has public repository on famous online hosting platform (or been hosted on npm)

# Auto-update checklist
 * [x] Has valid tags for each versions (for git auto-update)
 * [x] Auto-update setup
 * [x] Auto-update target/source is valid.
 * [x] Auto-update filemap is correct.

# Git commit checklist
 * [x] The first line of commit message is less then 50 chars, be clean and clear, easy to understand.
 * [x] The parent of the commit(s) in the PR is not old than 3 days.
 * [x] Pull request is sending from a non-master branch with meaningful name.
 * [x] Separate unrelated changes into different commits.
 * [x] Use rebase to squash/fixup dummy/unnecessary commits into only one commit.
 * [x] Close corresponding issue in commit message
 * [x] Mention related issue(s), people in commit message, comment.

close #8023, cc @gilbitron

### CODE REVIEW

**Comment by dannyxx001** [comment]
Sorry for late response. I think the PR is OK, we can rebase first then call Peter.

**Comment by kennynaoh** [comment]
@dannyxx001 
I have rebased to the latest master branch.
Please help me review it again.
Thanks!

**Comment by dannyxx001** [comment]
Sorry, I think we should rebase again. XD In fact, if next time I say the PR is OK except for rebasing it, then after you rebase it, you can directly call Peter.

**Comment by kennynaoh** [comment]
@dannyxx001 @PeterDaveHello 
I have rebased to the latest master branch.
Please help me review it again.
Thanks!

**Comment by PeterDaveHello** [comment]
Nope, every updates need its review, I only review the reviewed PRsby reviewer.

**Comment by PeterDaveHello** [comment]
Don't we need `extensions/` and `theme/`? Do not just add it because I'm asking you to discuss and think, don't make yourselves like the Brian you know.

**Comment by kennynaoh** [comment]
I usually discusses with other participants before I send a PR.
So if there are some difficult problems I can't solve, I just send it and discuss with you on-line.
And this case I think the main file are `ideal-image-slider.js` and `ideal-image-slider.css`.
I really don't know what's the reason why we need `extensions/` and `theme/`.
Sorry about that.
So whether I should ask author about that?
Thanks!

**Comment by PeterDaveHello** [comment]
Read its readme as it already been mentioned.
https://github.com/gilbitron/Ideal-Image-Slider#ideal-image-slider
https://github.com/gilbitron/Ideal-Image-Slider#getting-started

**Comment by kennynaoh** [comment]
Got it
@dannyxx001 
I have added `extensions/` and `themes/`.
Please help me review it again.
Thanks!

**Comment by dannyxx001** [comment]
Sorry for late response. I think the PR is OK, but we need to rebase first. After finish rebase, ping me again, thank you.

**Comment by kennynaoh** [comment]
ping @dannyxx001 
I have rebased to the latest master branch.
Please help me review it again.
Thanks!

File: ajax/libs/ideal-image-slider/package.json [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+    "url": "https://github.com/gilbitron/Ideal-Image-Slider.git"
+  },
+  "npmName": "ideal-image-slider",
+  "npmFileMap": [
+    {
+      "basePath": "",
+      "files": [
+        "ideal-image-slider*",
+        "extensions/**/*.!(md)",
+        "themes/**/*.!(md)"
```
Comment (PeterDaveHello) [inline ajax/libs/ideal-image-slider/package.json]: There is no `*.md` under `themes`, so why did you use this pattern?

**Comment by kennynaoh** [comment]
@dannyxx001 
I have fixed the npmFileMap.
Please help me review it again.
Thanks!

**Comment by dannyxx001** [comment]
Sorry for late response. I think the PR is OK, but we need to rebase first. :)

**Comment by kennynaoh** [comment]
@dannyxx001 
I have rebased to the latest master branch.
Please help me review it again.
Thanks!

**Comment by PeterDaveHello** [comment]
@kennynaoh actually you can merge the second and the third file map next time.

**Comment by PeterDaveHello** [comment]
@kennynaoh since there are only 3 versions on npm, please add the other versions from git repo in a new PR.

**Pull request merged**